lib/compiler/src/v3_core.erl

```diff
@@ -2025,10 +2025,18 @@ is_safe(_) -> false.
 %% fold_match(MatchExpr, Pat) -> {MatchPat,Expr}.
 %%  Fold nested matches into one match with aliased patterns.
 
-fold_match({match,L,P0,E0}, P) ->
-    {P1,E1} = fold_match(E0, P),
-    {{match,L,P0,P1},E1};
-fold_match(E, P) -> {P,E}.
+fold_match(P, E) ->
```

Comment:
Thanks! That **is** simpler and works if you add a clause to handle the case that the first argument is not a `match` tuple:

```
fold_match({match, L0, P0, E1}, E0) ->
    fold_match_helper(E1, {match, L0, P0, E0});
fold_match(P, E) ->
    {E, P}.
```

Staring some more at the code, I saw that it could be further simplified by removing `fold_match/2` and renaming the helper function to `fold_match/2`. I have pushed a fixup commit that does that.

Reply:
Perfect!